smarts/core/sumo_road_network.py

```diff
@@ -324,10 +324,15 @@ def oncoming_lanes_at_offset(self, offset: float) -> List[RoadMap.Lane]:
             if not nearby_lanes:
                 return result
             my_vect = self.vector_at_offset(offset)
-            my_norm = np.linalg.norm(my_vect)
+            my_norm = max(np.linalg.norm(my_vect), 1e-6)
             threshold = -0.995562  # cos(175*pi/180)
             for lane, _ in nearby_lanes:
-                lv = lane.vector_at_offset(offset)
+                if lane == self:
+                    continue
+                lane_refline_pt = lane.to_lane_coord(pt)
+                lv = lane.vector_at_offset(lane_refline_pt.s)
+                if np.array_equal(lv, np.array([0.0, 0.0, 0.0])):
```

Comment:
Perhaps a slightly nicer option here would be to first compute the norm and then compare it to 0, like:
```python
...
lv_norm = np.linalg.norm(lv)
if lv_norm == 0:
    continue
lane_angle = np.dot(my_vect, lv) / (my_norm * lv_norm)
...
````